src/shared/istiopolicy/creator.go

```diff
@@ -98,6 +102,80 @@ func (c *Creator) Create(
 	return nil
 }
 
+func (c *Creator) UpdateIntentsStatus(
+	ctx context.Context,
+	clientIntents *v1alpha2.ClientIntents,
+	clientServiceAccount string,
+) error {
+	err := c.labelWithServiceAccount(ctx, clientIntents, clientServiceAccount)
+	if err != nil {
+		return err
+	}
+
+	return c.updateStatusSharedServiceAccount(ctx, clientIntents.Namespace, clientServiceAccount)
+}
+
+func (c *Creator) labelWithServiceAccount(ctx context.Context, clientIntents *v1alpha2.ClientIntents, clientServiceAccount string) error {
+	serviceAccountLabelValue, ok := clientIntents.Labels[v1alpha2.OtterizeClientServiceAccountLabel]
+	if ok && serviceAccountLabelValue == clientServiceAccount {
+		return nil
+	}
+
+	labeledIntents := clientIntents.DeepCopy()
+	if labeledIntents.Labels == nil {
+		labeledIntents.Labels = make(map[string]string)
+	}
+
+	labeledIntents.Labels[v1alpha2.OtterizeClientServiceAccountLabel] = clientServiceAccount
+	err := c.client.Patch(ctx, labeledIntents, client.MergeFrom(clientIntents))
+	if err != nil {
+		logrus.WithError(err).Errorln("Failed labeling intent with service account name")
+		c.recorder.RecordWarningEventf(clientIntents, ReasonIntentsLabelFailed, "Failed labeling intent with service account name: %s", err.Error())
+		return err
+	}
+
+	logrus.Infof("Labeling intent %s with service account name %s", clientIntents.Name, clientServiceAccount)
+
+	c.recorder.RecordNormalEventf(clientIntents, ReasonServiceAccountFound, "Client intents service account found %s", clientServiceAccount)
+	return nil
+}
+
+func (c *Creator) updateStatusSharedServiceAccount(ctx context.Context, namespace string, clientServiceAccountName string) error {
+	var ClientIntentsList v1alpha2.ClientIntentsList
+	err := c.client.List(ctx, &ClientIntentsList, client.InNamespace(namespace), client.MatchingLabels{
+		v1alpha2.OtterizeClientServiceAccountLabel: clientServiceAccountName,
```

Comment:
Would this work with all possible values?
Kubernetes resource names are limited to 253 characters
Label values are limited to 63 characters
So please use an annotation